assets/global.js

```diff
@@ -879,7 +892,42 @@ class SlideshowComponent extends SliderComponent {
   }
 }
 
-customElements.define('slideshow-component', SlideshowComponent);
+customElements.define('slideshow-section-component', SlideshowSectionComponent);
+
+class AnnouncementBarSlider extends SlideshowSharedComponent {
+  constructor() {
+    super();
+
+    this.announcementBarArrowButtonWasClicked = false;
+
+    [this.reducedMotion, this.desktopLayout].forEach((mediaQuery) => {
+      mediaQuery.addEventListener('change', () => {
+        if (this.slider.getAttribute('data-autoplay') === 'true') this.setAutoPlay();
+      });
+    });
+
+    [this.prevButton, this.nextButton].forEach((button) => {
+      button.addEventListener('click', () => {
+        this.announcementBarArrowButtonWasClicked = true;
+      }, {once: true});
+    });
+  }
+
+  setAutoPlay() {
+    super.setAutoPlay();
+    this.reducedMotion.matches || this.announcementBarArrowButtonWasClicked || !this.desktopLayout.matches ? this.pause() : this.play();
+  }
+
+  focusOutHandling() {
+    if (!this.reducedMotion.matches && !this.announcementBarArrowButtonWasClicked && this.desktopLayout.matches) this.play();
+  }
+
+  focusInHandling(event) {
+    if (this.querySelector('.announcement-bar-slider').contains(event.target)) this.pause();
+  }
+}
```

Comment:
If these are not using the inherited `focusInHandling` and `focusOutHandling`, do we need them in the `SlideshowSharedComponent` class at all?

Reply:
I changed it as you suggested, thank you.